pymeasure/instruments/pendulum/cnt91.py

```diff
@@ -0,0 +1,191 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from time import sleep
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import (
+    strict_discrete_set,
+    strict_range,
+    truncated_range,
+)
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class CNT91(Instrument):
+    """Represents a Pendulum CNT-91 frequency counter."""
+
+    CHANNELS = {"A": 1, "B": 2, "C": 3, "E": 4, "INTREF": 6}
+    MAX_BUFFER_SIZE = 32000  # User Manual 8-38
+
+    def __init__(self, resourceName, **kwargs):
+        super().__init__(resourceName, "Pendulum CNT-91", **kwargs)
+        self.adapter.connection.timeout = 120000
+
+    def ask(self, command):
+        """
+        Writes the command to the instrument through the adapter
+        and returns the read response.
+
+        :param command: command string to be sent to the instrument
+        """
+        # Infinite loop occures if not redefined.
+        return super().ask(command).rstrip("\n")
+
+    @property
+    def operation_complete(self):
+        """Is True if operation is complete."""
+        return self.ask("*OPC?") == "1"
+
+    @property
+    def batch_size(self):
+        """Maximum number of buffer entries that can be transmitted at once."""
+        if not hasattr(self, "_batch_size"):
+            self._batch_size = int(self.ask("FORM:SMAX?"))
+        return self._batch_size
+
+    def read_buffer(self):
+        """
+        Read out the entire buffer.
+
+        :return : Frequency values from the buffer.
+        """
+        while not self.operation_complete:
+            # Wait until the buffer is filled.
+            sleep(0.01)
+        data = []
+        # Loop until the buffer is completely read out.
+        while True:
+            # Get maximum number of buffer values.
+            new = self.values(":FETC:ARR? MAX")
+            data += new
+            # Last values have been read from buffer.
+            if len(new) < self.batch_size:
+                break
+        return data
+
+    external_start_arming_source = Instrument.control(
+        "ARM:SOUR?",
+        "ARM:SOUR %s",
+        (
+            "Select arming input or switch off the start arming function."
+            "Options are 'A', 'B' and 'E' (rear). 'IMM' turns trigger off."
+        ),
+        validator=strict_discrete_set,
+        values={"A": "EXT1", "B": "EXT2", "E": "EXT4", "IMM": "IMM"},
+        map_values=True,
+    )
+
+    external_arming_start_slope = Instrument.control(
+        "ARM:SLOP?",
+        "ARM:SLOP %s",
+        "Set slope for the start arming condition.",
```

Comment:
This should be a docstring if I am not mistaken

Reply:
Do you mean a triple-quote delimited string `"""bla bla"""`? If so, that is only Python's way of defining an optionally multiline string. A normal string is fine here.

Reply:
Yes exactly.
I had some issues with the display of short docstrings in the documentation in the past and have defaulted to use triple-qouted docstrings with a new line at the end.

Reply:
I don't have a strong opinion either way. Changed to triple quotes since in 1704e41 since it seems a little bit more of the standard formatting and number of lines is the same.